src/controllers/UsersController.php

```diff
@@ -298,4 +298,36 @@ public function passwordReset(Request $request, $db)
         }
 
     }
+
+
+
+    public function deleteAction($request, $db)
+    {
+
+
+        if (! isset($request->user_id)) {
+            throw new Exception("You must be logged in to delete data", 400);
+        }
+        // delete the talk
+        $user_id     = $this->getItemId($request);
+        $user_mapper = new UserMapper($db, $request);
+
+        // note: use the mapper's getUserById as we don't want to throw a not found exception
+        $user = $user_mapper->getUserById($user_id);
+        if (false === $user) {
+            // user isn't there so it's as good as deleted
+            header("Content-Length: 0", null, 204);
```

Comment:
I don't think it's strange. It makes the DELETE action idempotent, as it should be. If, for example, a mobile app retries the DELETE action because it missed the response on the previous request, it won't show an error.